src/main/java/com/puppycrawl/tools/checkstyle/xpath/RootNode.java

```diff
@@ -143,9 +143,11 @@ public NodeInfo getRoot() {
      *
      * @param axisNumber element from {@code AxisInfo}
      * @return {@code AxisIterator} object
+     * @noinspection resource, IOResourceOpenedButNotSafelyClosed
```

Comment:
Please explain reason of suppression in javadoc

Reply:
@timurt there is a bug in the eclipse compiler related to this issue:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=383965
Please note this in the method Javadoc.

Reply:
@romani done

Reply:
@pbludov do I really need to specify this bug in javadoc, isn't it CI/CD problem?

Reply:
My point is that there is no need for suppression. But due to bugs, some ci tools reports a violation here. One day these bugs will be fixed and the suppression will become redundant. So I suggested mentioning the bug here.